### PR TITLE
fix(dev-deps): update ts-node and spawn

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR fixes how we spawn a child_process and bumps `ts-node`.

(H/T to @cspotcode for identifying this after we opened an issue in `ts-node`)

Related:
- https://github.com/TypeStrong/ts-node/issues/1283#issuecomment-807704228
- Reverts https://github.com/cdr/code-server/pull/2962
- https://github.com/cdr/code-server/pull/2965

image

### CODE REVIEW

File: src/node/wrapper.ts
Comment (code-asher) [inline src/node/wrapper.ts:317]: I'm thinking we could use `inherit` so we can get rid of the manual `pipe` calls below.

```suggestion
      stdio: ["inherit", "inherit", "inherit", "ipc"],
```

**Review by code-asher — CHANGES_REQUESTED** [review]
Thank you for taking care of this!

File: src/node/wrapper.ts
Comment (code-asher) [reply src/node/wrapper.ts]: Uh, so, this is embarrassing...I was looking at the wrong file. We could still use inherit but we'd need to remove the `pipe` calls to `process.stdout` and `process.stderr` and pipe them to the logger instead of the child. 

I say we just leave these at `pipe` for now haha. Sorry for the noise. :sob:

Comment (jsjoeio) [reply src/node/wrapper.ts]: LOL no worries at all! Glad you caught it before we merged. I'm less familiar with all this `pipe` stuff so kinda following your lead. Okay will do!

**Comment by jsjoeio** [comment]
bump @code-asher - ready for a second review

**Review by code-asher — APPROVED** [review]

**Pull request merged**
